src/AppInstallerCLICore/Workflows/UpdateFlow.cpp

```diff
@@ -190,19 +196,18 @@ namespace AppInstaller::CLI::Workflow
                 continue;
             }
 
-            // Filter out packages that require explicit upgrades.
-            // We require explicit upgrades only if the installed version is pinned,
-            // either because it was manually pinned or because the manifest indicated
-            // RequireExplicitUpgrade.
-            // Note that this does not consider whether the update to be installed has
-            // RequireExplicitUpgrade. While this has the downside of not working with
-            // packages installed from another source, it ensures consistency with the
-            // list of available updates (there we don't have the selected installer)
-            // and at most we will update each package like this once.
+            // Filter out packages that require explicit upgrade.
+            // User-defined pins are handled when selecting the version to use.
             auto installedMetadata = updateContext.Get<Execution::Data::InstalledPackageVersion>()->GetMetadata();
-            auto pinnedState = ConvertToPackagePinnedStateEnum(installedMetadata[PackageVersionMetadata::PinnedState]);
-            if (pinnedState != PackagePinnedState::NotPinned)
+            auto pinnedState = ConvertToPinTypeEnum(installedMetadata[PackageVersionMetadata::PinnedState]);
+            if (pinnedState == PinType::PinnedByManifest)
             {
+                // Note that for packages pinned by the manifest
+                // this does not consider whether the update to be installed has
+                // RequireExplicitUpgrade. While this has the downside of not working with
+                // packages installed from another source, it ensures consistency with the
+                // list of available updates (there we don't have the selected installer)
+                // and at most we will update each package like this once.

```

Comment:
Does this PR also resolve https://github.com/microsoft/winget-cli/issues/2559#issuecomment-1262712788 ?

Reply:
No, it does not. If we make any changes for that, I'd rather have them in a separate PR as this one is already huge.

Reply:
I understand :)
Just not sure where the original bug actually came in, so was hoping it might have been cauggt in the mix. Thank you for the clarification!